### ISSUE TITLE
Tensorflow build failed to complete

### ISSUE THREAD

**Issue by AjayZinngg** [tensorflow/tensorflow#23673]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): source
- TensorFlow version: 1.10
- Python version: 3
- Installed using virtualenv? pip? conda?: virtualenv/pip
- Bazel version (if compiling from source): 0.19.0
- GCC/Compiler version (if compiling from source): gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.10) 
- CUDA/cuDNN version:
- GPU model and memory:N/A

**Describe the problem**
I am building a mobile object detection app and trying to convert the model to tensorflow lite model. Accoding to the steps in [this](https://medium.com/tensorflow/training-and-serving-a-realtime-mobile-object-detector-in-30-minutes-with-cloud-tpus-b78971cf1193) blog post, I am supposed to build tensorflow from source using bazel and I followed steps [here](https://www.tensorflow.org/install/source). But after executing for sometime the build process fails. 

**Provide the exact sequence of commands / steps that you executed before running into the problem**
`git clone https://github.com/tensorflow/tensorflow.git
 cd tensorflow`
`bazel test -c opt -- //tensorflow/... -//tensorflow/compiler/... -//tensorflow/contrib/lite/...`

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

The error message is : 
```
ERROR: /home/stash/.cache/bazel/_bazel_stash/f5e22f69335a98eb99581e9c850e62a4/external/grpc/BUILD:1332:1: C++ compilation of rule '@grpc//:grpc_resolver_dns_ares' failed (Exit 1) gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -B/usr/bin -B/usr/bin -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG ... (remaining 53 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
external/grpc/src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_ev_driver_posix.cc:23:18: fatal error: ares.h: No such file or directory
compilation terminated.
INFO: Elapsed time: 26.272s, Critical Path: 18.50s, Remote (0.00% of the time): [queue: 0.00%, setup: 0.00%, process: 0.00%]
INFO: 12 processes: 12 linux-sandbox.
FAILED: Build did NOT complete successfully
//tensorflow/c:c_api_experimental_test                                NO STATUS
//tensorflow/c:c_api_function_test                                    NO STATUS
//tensorflow/c:c_api_test_gpu                                         NO STATUS
//tensorflow/c:while_loop_test                                        NO STATUS
//tensorflow/c/eager:c_api_test_gpu                                   NO STATUS
//tensorflow/cc:cc_op_gen_test                                        NO STATUS
//tensorflow/cc:client_client_session_test                            NO STATUS
...
...
//tensorflow/tools/proto_text:gen_proto_text_functions_lib_test       NO STATUS

FAILED: Build did NOT complete successfully
```

Elapsed time Critical Path are low in the error message as I had tried the command again after it failed in the first try.

**Comment by AjayZinngg** [comment]
After this command executes, I am supposed to run this command according to the blog [post](https://medium.com/tensorflow/training-and-serving-a-realtime-mobile-object-detector-in-30-minutes-with-cloud-tpus-b78971cf1193):
`bazel run -c opt tensorflow/contrib/lite/toco:toco -- \
--input_file=$OUTPUT_DIR/tflite_graph.pb \
--output_file=$OUTPUT_DIR/detect.tflite \
--input_shapes=1,300,300,3 \
--input_arrays=normalized_input_image_tensor \
--output_arrays='TFLite_Detection_PostProcess','TFLite_Detection_PostProcess:1','TFLite_Detection_PostProcess:2','TFLite_Detection_PostProcess:3'  \
--inference_type=QUANTIZED_UINT8 \
--mean_values=128 \
--std_values=128 \
--change_concat_input_ranges=false \
--allow_custom_ops`

**Comment by bjascob** [comment]
It looks to me like you have an issue with your gcc configuration (maybe gcc isn't in /usr/bin/gcc??).   The command `bazel test` is running the unit tests with some default options.  I've had trouble running this command verbatim and noticed that they don't ask you to run `./configure` before running this.  I'm not sure if that's an oversight or if it's not required but when you run `./configure` you create a file named `.tf_configure.bazelrc` where things like the location of gcc is configured.

`bazel test` is optional.  You may have better luck skipping these unit tests and go directly to the `bazel build` step.  For me, `test` also has a lot of errors which look like they're related the test setup rather than the actual code itself.  My Tensorflow runs fine when simply doing the configure/build steps.

**Comment by NorwayLobster** [comment]
same problem, how to solve it?

**Comment by forrestrice** [comment]
I encountered the same issue with Tensorflow releases 1.11 and 1.12

**Comment by wt-huang** [comment]
Try to use bazel 0.15, gcc 4.8, TensorFlow 1.12. Post any errors you observe.

**Comment by NVchargr** [comment]
centos 6, gcc 5.5.0, bazel 0.19.0, tensorflow 1.12.0

In tensorflow 1.10.0 I find:
```
build --define=grpc_no_ares=true
```
in `.tf_configure.bazelrc` after running configure which comes from:
```
configure.py:  write_to_bazelrc('build --define grpc_no_ares=true')
```

The configure in 1.12.0 does not do this and tools/bazel.rc is not used with bazel 0.19.0

When starting the build I see:
```
WARNING: The following rc files are no longer being read, please transfer their contents or import their path into one of the standard rc files:
/build/tmp/tensorflow/tools/bazel.rc
```

So I added `import /build/tmp/tensorflow/tools/bazel.rc` to `.bazelrc`.

I am not sure if this is the "right" solution, but may shed some light on where the problem lies.

Perhaps `bazel test` does not load/read `tools/bazel.rc`?
In newer bazel versions `tools/bazel.rc` is not read at all.

**Comment by erley** [comment]
On my Debian GNU/Linux 9.6 (stretch) I installed prerequisities:
- nvidia-driver 390.87 (official repo)
- Nvidia CUDA toolkit 9.1.85 + libcublas9.1 (official repo)
- libcudnn7 7.1.3.16-1 for cuda9.1 (via deb file)
- libnccl 2.1.15 for cuda 9.1 (via deb file)
- bazel 0.19.2 (official google repo)
- and usual python libs via pip

Then cloned Tensorflow sources, switched to its **r1.12** branch, run **./configure** and added line

`import /build/tmp/tensorflow/tools/bazel.rc`

to **.bazelrc**
Compilation went well with the usual command

`bazel build --config=opt --config=cuda //tensorflow/tools/pip_package:build_pip_package`

It correctly detects and runs on my GTX1060 GPU.
Hope it helps!

**Comment by metalalive** [comment]
I got the same  error, build from source in the following environment  

#### system information
- OS Platform and Distribution : Raspbian  9.0 (Stretch)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): source
- TensorFlow version: 1.12
- Python version: N/A, I was building C++ library (.so)
- Installed using virtualenv? pip? conda?: N/A
- Bazel version (if compiling from source): 0.19.2
- GCC/Compiler version (if compiling from source): gcc version 6.3.0
- CUDA/cuDNN version: N/A
- GPU model and memory:N/A

#### the exact sequence of commands / steps that you executed before running into the problem
```
git clone https://github.com/tensorflow/tensorflow.git
cd tensorflow
git checkout r1.12
./configure
bazel build -c opt --copt="-mfpu=neon-vfpv4" --copt="-funsafe-math-optimizations" --copt="-ftree-vectorize" --copt="-fomit-frame-pointer" --local_resources 1024,1.0,1.0 --verbose_failures //tensorflow:libtensorflow_cc.so
```

**Comment by jvishnuvardhan** [comment]
Thanks @erley and @metalalive for providing solution. I think it was resolved.  Please open a new ticket if you see similar issue. Thanks!
